src/controller/c/supervisor.c

```diff
@@ -2055,6 +2083,61 @@ const char *wb_supervisor_node_get_base_type_name(WbNodeRef node) {
 
   return wb_node_get_name(node->type);
 }
+WbFieldRef wb_supervisor_node_get_field_by_index(WbNodeRef node, const int id) {
+  if (!robot_check_supervisor(__FUNCTION__))
+    return NULL;
+
+  if (!is_node_ref_valid(node)) {
```

Comment:
You should also check if the field index is valid, i.e. at least that index is >= 0.